appmap/_implementation/testing_framework.py

```diff
@@ -124,4 +126,11 @@ def record(self, klass, method, **kwds):
         filename = fi.filename + '.appmap.json'
         metadata = fi.metadata
         metadata.update(self.metadata)
-        (self.appmap_path / filename).write_text(generation.dump(rec, metadata))
+        appmap_json = self.appmap_path / filename
+        with NamedTemporaryFile(mode='w', dir=self.appmap_path, delete=False) as f:
+            f.write(generation.dump(rec, metadata))
+            try:
+                os.remove(appmap_json)
+            except FileNotFoundError:
+                pass
+            os.replace(f.name, appmap_json)
```

Comment:
This is not guaranteed to work correctly on every operating system, since the file is still open at this point. Either close it first explicitly or move the move out of the context.